### ISSUE TITLE
[Codegen] BatchNorm compilation issue

### ISSUE THREAD

**Issue by IanWood1** [iree-org/iree#21616]
### What happened?

For context, I made some changes to dispatch creation to improve fusion for batch norm but ended up not being able to get it to fully compile due to long compile times. One of the two dispatches from a BatchNorm2d program ends up getting very large vector sizes causing a _very_ long compile time & probably very poor performance. I disabled warp reduction and it was able to compile but the full program got 4x slower when comparing to WarpReduction + no fusion changes.

[Executable source for the problematic dispatch](https://gist.github.com/IanWood1/3c0b514c9655a75d2877e4c047164d81)
[Input MLIR](https://gist.githubusercontent.com/zjgarvey/eb925cb7f9136794bed9b6e1a2c6c0b3/raw/daeb280f3c5df725b27591a54fede7b7164d3eb6/sample_ir.mlir)
[Full program after forming dispatches (with my changes)](https://gist.github.com/IanWood1/c9a2adccd6b9a469535da068d7b19faf)

### Steps to reproduce your issue

1. Download the [executable source](https://gist.github.com/IanWood1/3c0b514c9655a75d2877e4c047164d81)
2. Run `iree-compile repro.mlir`

### Version information

35a872c17908f7e459fdebd8cbc813128e37ad56

**Comment by IanWood1** [comment]
@newling I think this relates to the work you are doing to remove warp reduction.

**Comment by IanWood1** [comment]
Here's the same dispatch but without the transposes on the indexing maps: https://gist.github.com/IanWood1/db18f1a5381837bb31378cd702b2b9a9. It still has the same issue

**Comment by Groverkss** [comment]
I don't see any long compile times with this IR anymore with VectorDistribute. Please reopen if there are issues.
